observability/src/test/java/io/grpc/observability/interceptors/InternalLoggingChannelInterceptorTest.java

```diff
@@ -0,0 +1,331 @@
+          expectedResponseMessageEvent);
+      verifyNoMoreInteractions(mockSink);
+      verify(mockListener).onMessage(same(response));
+    }
+
+    // receive trailer
+    {
+      EventType expectedTrailerEvent = EventType.GRPC_CALL_TRAILER;
+      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
+      Status status = Status.INTERNAL.withDescription("trailer description");
+      Metadata trailers = new Metadata();
+
+      interceptedListener.get().onClose(status, trailers);
+      verify(mockSink, times(6)).write(captor.capture());
+      assertEquals(captor.getValue().getEventType(),
+          expectedTrailerEvent);
+      verifyNoMoreInteractions(mockSink);
+      verify(mockListener).onClose(same(status), same(trailers));
+    }
+
+    // cancel
+    {
+      EventType expectedCancelEvent = EventType.GRPC_CALL_CANCEL;
+      ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
+      interceptedLoggingCall.cancel(null, null);
+      verify(mockSink, times(7)).write(captor.capture());
+      assertEquals(captor.getValue().getEventType(),
+          expectedCancelEvent);
+      cancelCalled.get(1, TimeUnit.SECONDS);
+    }
+  }
+
+  @Test
+  public void clientDeadLineLogged_deadlineSetViaCallOption() {
+    MethodDescriptor<byte[], byte[]> method =
+        MethodDescriptor.<byte[], byte[]>newBuilder()
+            .setType(MethodType.UNKNOWN)
+            .setFullMethodName("service/method")
+            .setRequestMarshaller(BYTEARRAY_MARSHALLER)
+            .setResponseMarshaller(BYTEARRAY_MARSHALLER)
+            .build();
+    @SuppressWarnings("unchecked")
+    ClientCall.Listener<byte[]> mockListener = mock(ClientCall.Listener.class);
+
+    ClientCall<byte[], byte[]> interceptedLoggingCall =
+        factory.create()
+            .interceptCall(
+                method,
+                CallOptions.DEFAULT.withDeadlineAfter(1, TimeUnit.SECONDS),
+                new Channel() {
+                  @Override
+                  public <RequestT, ResponseT> ClientCall<RequestT, ResponseT> newCall(
+                      MethodDescriptor<RequestT, ResponseT> methodDescriptor,
+                      CallOptions callOptions) {
+                    return new NoopClientCall<>();
+                  }
+
+                  @Override
+                  public String authority() {
+                    return "the-authority";
+                  }
+                });
+    interceptedLoggingCall.start(mockListener, new Metadata());
+    ArgumentCaptor<GrpcLogRecord> captor = ArgumentCaptor.forClass(GrpcLogRecord.class);
+    verify(mockSink, times(1)).write(captor.capture());
+    Duration timeout = captor.getValue().getTimeout();
+    assertThat(TimeUnit.SECONDS.toNanos(1) - Durations.toNanos(timeout))
+        .isAtMost(TimeUnit.MILLISECONDS.toNanos(250));
+  }
+
+  @Test
+  public void clientDeadlineLogged_deadlineSetViaContext() throws Exception {
```

Comment:
what about when both are set? You have to use the min - is that being tested?

Reply:
I guess this is addressed? resolving the conversation?

Reply:
Yes, added test case `clientDeadlineLogged_deadlineSetViaContextAndCallOptions()` to check when both are set.